pkg/stats.go

```diff
@@ -8,44 +8,123 @@ import (
+	// not including inputs and outputs.
+	TotalDescriptionBytes int
+
+	// TotalInputPropertyDescriptionBytes is the sum total of all bytes in descriptions of function input properties,
+	// not including the input type description.
+	TotalInputPropertyDescriptionBytes int
+
+	// InputPropertiesMissingDescriptions is the total number of all function input properties missing descriptions.
+	InputPropertiesMissingDescriptions int
+
+	// TotalOutputPropertyDescriptionBytes is the sum total of all bytes in description of function output properties,
+	// not include the output type description.
+	TotalOutputPropertyDescriptionBytes int
+
+	// OutputPropertiesMissingDescriptions is the total number of all function output properties missing descriptions.
+	OutputPropertiesMissingDescriptions int
 }
 
 func CountStats(sch schema.PackageSpec) PulumiSchemaStats {
-	stats := PulumiSchemaStats{}
+	stats := PulumiSchemaStats{
+		Resources: ResourceStats{},
+		Functions: FunctionStats{},
+	}
 
 	uniques := codegen.NewStringSet()
 	visitedTypes := codegen.NewStringSet()
 
-	var propCount func(string) (int, int)
-	propCount = func(typeName string) (totalProperties int, propertiesMissingDesc int) {
+	type propCountResult struct {
+		totalInputs        int
+		inputsMissingDesc  int
+		totalOutputs       int
+		outputsMissingDesc int
+	}
+
+	var propCount func(string) propCountResult
+	propCount = func(typeName string) propCountResult {
 		if visitedTypes.Has(typeName) {
-			return 0, 0
+			return propCountResult{}
 		}
 
+		res := propCountResult{}
+
 		visitedTypes.Add(typeName)
 
 		t := sch.Types[typeName]
 
-		totalProperties = len(t.Properties)
-		propertiesMissingDesc = 0
+		res.totalInputs = len(t.Properties)
 
-		for _, p := range t.Properties {
-			if p.Description == "" {
-				propertiesMissingDesc++
+		for _, input := range t.Properties {
+			if input.Description == "" {
+				res.inputsMissingDesc++
 			}
 
-			if p.Ref != "" {
-				tn := strings.TrimPrefix(p.Ref, "#/types/")
-				nestedTotalProps, nestedPropsMissingDesc := propCount(tn)
-				totalProperties += nestedTotalProps
-				propertiesMissingDesc += nestedPropsMissingDesc
+			if input.Ref != "" {
+				tn := strings.TrimPrefix(input.Ref, "#/types/")
+				nestedRes := propCount(tn)
+
+				res.totalInputs += nestedRes.totalInputs
+				res.totalOutputs += nestedRes.totalOutputs
+				res.inputsMissingDesc += nestedRes.inputsMissingDesc
+				res.outputsMissingDesc += nestedRes.outputsMissingDesc
 			}
 		}
-		return totalProperties, propertiesMissingDesc
+
+		res.totalOutputs = len(t.ObjectTypeSpec.Properties)
+
+		for _, output := range t.ObjectTypeSpec.Properties {
+			if output.Description == "" {
+				res.outputsMissingDesc++
+			}
+
+			// TODO: Determine if outputs linking to a type is an actual schema capability
+			if output.Ref != "" {
+				tn := strings.TrimPrefix(output.Ref, "#/types/")
+				nestedRes := propCount(tn)
```

Comment:
etc.

Basically, `.Ref` types could be anywhere, or even one of our "well-known" types like `pulumi.json#/Asset`